src/sunnonlinsol/petscsnes/sunnonlinsol_petscsnes.c

```diff
@@ -269,98 +269,99 @@ int SUNNonlinSolFree_PetscSNES(SUNNonlinearSolver NLS)
 
   /* check that the nonlinear system function is non-null */
-  if (SysFn == NULL) { return (SUN_NLS_ILL_INPUT); }
+  if (SysFn == NULL) { return (SUN_ERR_ARG_INCOMPATIBLE); }
 
   SUNNLS_SNES_CONTENT(NLS)->Sys = SysFn;
-  return SUN_NLS_SUCCESS;
+  return SUN_SUCCESS;
 }
 
 /*==============================================================================
   Get functions
   ============================================================================*/
 
 /* get the PETSc SNES context underneath the SUNNonlinearSolver object */
-int SUNNonlinSolGetSNES_PetscSNES(SUNNonlinearSolver NLS, SNES* snes)
+SUNErrCode SUNNonlinSolGetSNES_PetscSNES(SUNNonlinearSolver NLS, SNES* snes)
 {
   /* check that the nonlinear solver is non-null */
-  if (NLS == NULL) { return SUN_NLS_MEM_NULL; }
+  if (NLS == NULL) { return SUN_ERR_ARG_CORRUPT; }
 
   /* return the SNES context */
   *snes = SUNNLS_SNESOBJ(NLS);
-  return SUN_NLS_SUCCESS;
+  return SUN_SUCCESS;
 }
 
 /* get the last error return by SNES */
-int SUNNonlinSolGetPetscError_PetscSNES(SUNNonlinearSolver NLS,
-                                        PetscErrorCode* err)
+SUNErrCode SUNNonlinSolGetPetscError_PetscSNES(SUNNonlinearSolver NLS,
+                                               PetscErrorCode* err)
 {
   /* check that the nonlinear solver is non-null */
-  if (NLS == NULL) { return SUN_NLS_MEM_NULL; }
+  if (NLS == NULL) { return SUN_ERR_ARG_CORRUPT; }
 
   /* return the last PETSc error code returned by SNES */
   *err = SUNNLS_SNES_CONTENT(NLS)->petsc_last_err;
-  return SUN_NLS_SUCCESS;
+  return SUN_SUCCESS;
 }
 
 /* get a pointer to the SUNDIALS integrator-provided system function F(y) */
-int SUNNonlinSolGetSysFn_PetscSNES(SUNNonlinearSolver NLS,
-                                   SUNNonlinSolSysFn* SysFn)
+SUNErrCode SUNNonlinSolGetSysFn_PetscSNES(SUNNonlinearSolver NLS,
+                                          SUNNonlinSolSysFn* SysFn)
 {
   /* check that the nonlinear solver is non-null */
-  if (NLS == NULL) { return SUN_NLS_MEM_NULL; }
+  if (NLS == NULL) { return SUN_ERR_ARG_CORRUPT; }
 
   /* return the nonlinear system defining function */
   *SysFn = SUNNLS_SNES_CONTENT(NLS)->Sys;
-  return SUN_NLS_SUCCESS;
+  return SUN_SUCCESS;
 }
 
 /* get the number of iterations performed in the last solve */
-int SUNNonlinSolGetNumIters_PetscSNES(SUNNonlinearSolver NLS, long int* nni)
+SUNErrCode SUNNonlinSolGetNumIters_PetscSNES(SUNNonlinearSolver NLS, long int* nni)
 {
   int ierr;
   sunindextype niters;
 
   /* check that the nonlinear solver is non-null */
-  if (NLS == NULL) { return SUN_NLS_MEM_NULL; }
+  if (NLS == NULL) { return SUN_ERR_ARG_CORRUPT; }
 
   /* get iteration count */
   ierr = SNESGetIterationNumber(SUNNLS_SNESOBJ(NLS), &niters);
   if (ierr != 0)
   {
     SUNNLS_SNES_CONTENT(NLS)->petsc_last_err = ierr;
-    return SUN_NLS_EXT_FAIL; /* ierr != 0 is not recoverable with PETSc */
+    return SUN_ERR_OP_FAIL; /* ierr != 0 is not recoverable with PETSc */
```

Comment:
I don't recall the exact name, but I thought that this PR also added a specific error flag to indicate an error in an external package.  If so, should that be used instead of `SUN_ERR_OP_FAIL`?